### PR TITLE
Add support for flake8 and pylint

### PR DESCRIPTION
**Pull request by jreese**
Summary:
- Adds two new run steps, one each for flake8 and pylint.
- Adds respective `run_flake8` and `run_pylint` config options.
- Adds a heavily customized `.pylint` config to the repo and fixes a few outstanding lint errors.
- Includes unit tests and updates existing tests to pass

Test Plan:
- Run `ptr` and it passes
- Introduce lint errors
- Run `ptr` and it fails with lint errors

Issue: #20

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -21,6 +21,10 @@
     "run_black": True,
     # Run mypy or not
     "run_mypy": True,
+    # Run flake8 or not
+    "run_flake8": True,
+    # Run flake8 or not
```
Comment (cooperlees) [inline setup.py]: NIT: pylint

File: ptr_tests.py [hunk 2]
```diff
@@ -200,6 +200,44 @@ def test_generate_mypy_cmd(self) -> None:
...
+        flake8_exe = Path("/bin/flake8")
+        with TemporaryDirectory() as td:
+            module_dir = Path(td)
+            subdir = module_dir / "awlib"
+            subdir.mkdir()
+            cf = module_dir / ".flake8"
+            py2 = subdir / "awesome2.py"
+            py1 = module_dir / "awesome.py"
+            for afile in (cf, py1, py2):
+                afile.touch()
```
Comment (cooperlees) [inline ptr_tests.py]: Maybe pass in cf to a function that creates py2 + py3 now we use it many times.

File: ci.py [hunk 3]
```diff
@@ -14,7 +14,7 @@
 from os import environ
 from pathlib import Path
 from subprocess import PIPE, run
-from sys import exit, stderr
+from sys import stderr
```
Comment (cooperlees) [inline ci.py]: `exit` is used on line 79 ... Which linter recommended this?

Comment (jreese) [reply ci.py]: `exit` is already available as a builtin, and does what `sys.exit` does.  pylint rightly points out that this was shadowing/redefining a builtin.  https://docs.python.org/3/library/constants.html#quit

**Comment by jreese** [comment]
Updated to fix nits, but now the ptr integration builds failed, and I'm going to bed.

**Comment by cooperlees** [comment]
Wow - 3.7 fails Integration, 3.6 does not:
```
/home/travis/build/facebookincubator/ptr/setup.py:
************* Module ptr_tests
/home/travis/build/facebookincubator/ptr/ptr_tests.py:18:0: W0611: Unused Dict imported from typing (unused-import)
/home/travis/build/facebookincubator/ptr/ptr_tests.py:18:0: W0611: Unused List imported from typing (unused-import)
```

Did this not fail locally too?

**Comment by jreese** [comment]
I tested locally on 3.6, just haven't had time to try it with 3.7 and see what's what.  Not sure why pylint would complain about it on one version and not another...

**Pull request merged**